src/rubrix/client/datasets.py

```diff
@@ -802,44 +809,65 @@ def prepare_for_training(self) -> "datasets.Dataset":
 
 
         """
-        import datasets
+        if framework is None:
 
-        has_annotations = False
-        for rec in self._records:
-            if rec.annotation is not None:
-                has_annotations = True
-                break
-
-        if not has_annotations:
-            return datasets.Dataset.from_dict({})
-
-        class_tags = ["O"]
-        class_tags.extend(
-            [
-                f"{pre}-{label}"
-                for label in sorted(self.__all_labels__())
-                for pre in ["B", "I"]
-            ]
-        )
-        class_tags = datasets.ClassLabel(names=class_tags)
+            import datasets
 
-        def spans2iob(example):
-            r = TokenClassificationRecord(
-                text=example["text"],
-                tokens=example["tokens"],
-                annotation=self.__entities_to_tuple__(example["annotation"]),
+            has_annotations = False
+            for rec in self._records:
+                if rec.annotation is not None:
+                    has_annotations = True
+                    break
+
+            if not has_annotations:
+                return datasets.Dataset.from_dict({})
+
+            class_tags = ["O"]
+            class_tags.extend(
+                [
+                    f"{pre}-{label}"
+                    for label in sorted(self.__all_labels__())
+                    for pre in ["B", "I"]
+                ]
             )
-            return class_tags.str2int(r.spans2iob(r.annotation))
+            class_tags = datasets.ClassLabel(names=class_tags)
 
-        ds = (
-            self.to_datasets()
-            .filter(self.__only_annotations__)
-            .map(lambda example: {"ner_tags": spans2iob(example)})
-        )
-        new_features = ds.features.copy()
-        new_features["ner_tags"] = [class_tags]
+            def spans2iob(example):
+                r = TokenClassificationRecord(
+                    text=example["text"],
+                    tokens=example["tokens"],
+                    annotation=self.__entities_to_tuple__(example["annotation"]),
+                )
+                return class_tags.str2int(r.spans2iob(r.annotation))
+
+            ds = (
+                self.to_datasets()
+                .filter(self.__only_annotations__)
+                .map(lambda example: {"ner_tags": spans2iob(example)})
+            )
+            new_features = ds.features.copy()
+            new_features["ner_tags"] = [class_tags]
+
+            return ds.cast(new_features)
+
+        elif framework == "spacy":
+
+            nlp = spacy.blank("en")
+
+            db = DocBin()
+
+            # Creating the docbin object as in https://spacy.io/usage/training#training-data
+            for record in self._records:
+                doc = nlp(record.text)
+                ents = []
+                for anno in record.annotation:
```

Comment:
As for the HF version, we should work only with annotated records. See `__only_annotations__` filter.

Reply:
With the HF version do you mean what it is now under `if framework is None`. I didn't know it was the Hugginface framework, I can name it that way and make it default. 

Then, should we only proceed with the method if there are annotations in the records? In both cases or only in HuggingFace? I think I didn't understand it properly.

Reply:
Yes. The training data should be generated only with records containing annotation. So, the record list should be filtered in that way as the default path

Reply:
I have added this, taken from the HF version

```
has_annotations = False
            for rec in self._records:
                if rec.annotation is not None:
                    has_annotations = True
                    break
```

Reply:
I've left some suggestion and comments